### ISSUE TITLE
matplotlib.pyplot.scatter should accept "markersize" for consistency

### ISSUE THREAD

**Issue by adrn** [matplotlib/matplotlib#1101]
I find it confusing that scatter() accepts the keyword argument "s" instead of markersize / ms like plot(). Was this a design choice? If so, why is "size" an invalid option? "s" is simply too ambiguous when other functions such as text() accept an "s" keyword which means string.

**Comment by story645** [comment]
does dealing with this simultaneously deal with #1099? (they're sort of addressing the same issue from two different sides)

**Comment by efiring** [comment]
Yes, any action dealing with either should address both of them.

**Comment by story645** [comment]
so at that point are they dupes?

**Comment by efiring** [comment]
Not exactly.  I think the thing to do is leave them both open.  Your comment links them, so that anyone wanting to work on one will see the discussion on the other.

**Comment by chahak13** [comment]
On the call today we discussed this and reached a consensus that this should be worked on. An idea discussed to start out with was to add a kwarg that would set the marker sizes in linear scale instead of the quadratic being done right now and if both `s` and `markersize` (or whatever we decide to call it) are provided then we error out as quickly with a detailed error message about it. The current state of action will be to first check the feasibility of implementing this. If this starts becoming unnecessarily complicated, we will rethink this.

cc: @tacaswell @efiring @jklymak @ksunden @melissawm @timhoffm @rcomer (Tim, tagging you since this would entail an API change probably)

**Comment by chahak13** [comment]
I think we need to decide on if we want to make this just a thing for `scatter` or an inherent property of `_CollectionWithSizes`. If we want to do it just for scatter then we can work around by handling the kwarg in the scatter function and "setting" the sizes as the squared value of the kwarg but this will create an inconsistency between what was set and what would be returned/used `get_sizes`/`set_sizes`. 

On the other hand, we can create a new `scaling` (placeholder name) attribute for `_CollectionWithSizes` that determines if the size was set with a linear scale or a quadratic scale and then sets the sizes with/without the `np.sqrt(self._sizes)` call. This way it gets handled by `_CollectionWithSizes` and can be extended to all the derived classes. I have a toy implementation for this if it sounds like something we want to do. It requires changes in `_axes.py`, `collections.py`, and `mplot3d/art3d.py` to handle 3D scatter.
